### DISCUSSION TITLE
How to get the expression matrix with celltype annoation from Seurat Object

### DISCUSSION THREAD

**Discussion by ryrl9703** [satijalab/seurat#5096] [Q&A]
I want to tranform the seurat object into a expression matrix with celltypes in column name and the gene symobl in row names or the first column, and now I have annoate the cell type. How should l get the expression matrix i want?
![微信图片_20210917181422](https://user-images.githubusercontent.com/87462884/133766708-3f87aee5-a62d-4b34-bdad-1597ec177c68.png)

**Comment by saketkc** [comment]
The data here looks already normalized. You can read such files as `read.delim("file.tsv", row.names=1, header=T)`. Though I would recommend starting with counts if you have them available.

**Reply by ryrl9703** [reply]
> The data here looks already normalized. You can read such files as `read.delim("file.tsv", row.names=1, header=T)`. Though I would recommend starting with counts if you have them available.

Thank you very much, but what I want is to get data from the seurat object like the data in the picture

**Comment by timoast** [comment]
You can use the `AverageExpression()` function to find the average expression of each gene for each cell type annotated in the dataset
